src/custom/hooks/useUSDCPrice/index.ts

```diff
@@ -0,0 +1,88 @@
+import { CurrencyAmount, Currency, Price, Token } from '@uniswap/sdk-core'
+import { useEffect, useMemo, useState } from 'react'
+import { unstable_batchedUpdates as batchedUpdate } from 'react-dom'
+import { supportedChainId } from 'utils/supportedChainId'
+import { useActiveWeb3React } from 'hooks/web3'
+import { getBestPrice } from 'utils/price'
+import { STABLECOIN_AMOUNT_OUT } from 'hooks/useUSDCPrice'
+import { stringToCurrency } from 'state/swap/extension'
+
+export * from '@src/hooks/useUSDCPrice'
+export { default } from '@src/hooks/useUSDCPrice'
+
+export function useBestUSDCPrice(currency?: Currency) {
+  const [bestUsdPrice, setBestUsdPrice] = useState<Price<Token, Currency> | null>(null)
+  const [error, setError] = useState<Error | null>(null)
+
+  const { chainId } = useActiveWeb3React()
+
+  // USDC constants
+  // 100_000e6 USDC @ 6 decimals
+  const amountOut = chainId ? STABLECOIN_AMOUNT_OUT[chainId] : undefined
+  const stablecoin = amountOut?.currency
+
+  useEffect(() => {
+    const isSupportedChain = supportedChainId(chainId)
+    if (!isSupportedChain || !currency || !amountOut || !stablecoin) return
+
+    const params = {
+      baseToken: stablecoin.address,
+      quoteToken: currency.wrapped.address,
+      kind: 'buy',
+      amount: amountOut.quotient.toString(),
+      chainId: isSupportedChain,
+      fromDecimals: currency.decimals,
+      toDecimals: stablecoin.decimals,
+    }
+
+    getBestPrice(params, { aggrOverride: 'max' })
```

Comment:
Why max? if this is a buy order you want to get the best price, this is the `min` sell token to buy those tokens. 

nit: Probably would be nice to have it in the PR description why you need to aggregate differently in this case. Probably u had your reasons

Reply:
discussing in slack, still unresolved

Reply:
ok we resovled it. I am leaving it to the original logic. But seeing as the useUSDCPrice hook runs an exact OUT buy order for USDC it will always take the lower price for both tokens